### PR TITLE
[zycore] Add new port 

### PR DESCRIPTION
**Pull request by fre3dm4n**
**Describe the pull request**

- #### What does your PR fix?
1. Add new port [zycore], which is a dependency of port [zydis]

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
all ,Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/zycore/fix-install.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
++++ b/CMakeLists.txt
+@@ -207,8 +207,11 @@ install(FILES
+ )
+ 
+ install(TARGETS "Zycore" EXPORT "zycore-targets"
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
++    ARCHIVE
++    LIBRARY
++    RUNTIME
```
Comment (JonLiu1993) [inline ports/zycore/fix-install.patch:13]:
```suggestion
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```

Comment (JonLiu1993) [reply ports/zycore/fix-install.patch:13]: @r4ve1, Thanks for your pr, could you please clarify why you remove this code?

File: ports/zycore/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+vcpkg_cmake_install()
+
+
+vcpkg_cmake_config_fixup(
+    CONFIG_PATH lib/cmake/${PORT}
+)
+    
+configure_file(${SOURCE_PATH}/LICENSE "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```
Comment (JonLiu1993) [inline ports/zycore/portfile.cmake:27]: Please add a blank line below the last line.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/zycore/portfile.cmake (hunk 2 above)
Comment (r4ve1) [reply ports/zycore/portfile.cmake]: Done

**Comment by r4ve1** [comment]
Seems I submitted some redundant commits... Forgive me, I'm new to vcpkg😢.

File: ports/zycore/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "zycore",
+  "version": "1.3.0",
+  "description": "Internal library of Zydis",
```
Comment (flobernd) [inline ports/zycore/vcpkg.json:4]:
```suggestion
  "description": "Provides platform independent types, macros and a fallback for environments without LibC.",
```

File: ports/zycore/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO zyantific/zycore-c
+    REF  4a8b5e2ab4d6ee73aa92d04bc318fed607394e67
+    SHA512 c707f5e07411d9f00fa59e3c382345009f225ed9406063b9863604f15a9c45c9a32bc9c3100f08d9c5800cc2254f71bfae817979b85bc604739ca1ee854c94e5
+    HEAD_REF master
+    PATCHES
+        fix-install.patch
+)
+
+string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} dynamic ZYCORE_BUILD_STATIC)
```
Comment (JonLiu1993) [inline ports/zycore/portfile.cmake:11]:
```suggestion
string(COMPARE EQUAL ${VCPKG_LIBRARY_LINKAGE} dynamic ZYCORE_BUILD_SHARED)

```

File: ports/zycore/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS -DZYCORE_BUILD_SHARED_LIB=${ZYCORE_BUILD_STATIC}
```
Comment (JonLiu1993) [inline ports/zycore/portfile.cmake:15]:
```suggestion
    OPTIONS -DZYCORE_BUILD_SHARED_LIB=${ZYCORE_BUILD_SHARED}

```

Comment (JonLiu1993) [reply ports/zycore/portfile.cmake:15]: Although the previous naming does not affect usage, it is easy to cause misunderstandings

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
Tetsed zycore:x64-windows usage successfully.

File: ports/zycore/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "zycore",
+  "version": "1.3.0",
+  "description": "Provides platform independent types, macros and a fallback for environments without LibC.",
```
Comment (dg0yt) [inline ports/zycore/vcpkg.json:4]: Is there a risk of symbol clash when used together with other libraries on windows when users rely on autolinking in certains MSVC usage scenarios?

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/zycore/vcpkg.json (hunk 6 above)
Comment (r4ve1) [reply ports/zycore/vcpkg.json:4]: You mean another package named the same as [zycore]? As far as I know there is not.

Comment (dg0yt) [reply ports/zycore/vcpkg.json:4]: No, I mean the symbol names provided by the library.

Comment (r4ve1) [reply ports/zycore/vcpkg.json:4]: Well... I think only author himself can answer this 🤣 @flobernd

File: ports/zycore/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+vcpkg_cmake_install()
+
+
+vcpkg_cmake_config_fixup(
+    CONFIG_PATH lib/cmake/${PORT}
+)
+    
+configure_file(${SOURCE_PATH}/LICENSE "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" COPYONLY)
```
[the remark this answers is not in the document]
Comment (r4ve1) [reply ports/zycore/portfile.cmake]: Done

File: ports/zycore/fix-install.patch [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
++++ b/CMakeLists.txt
+@@ -207,8 +207,11 @@ install(FILES
+ )
+ 
+ install(TARGETS "Zycore" EXPORT "zycore-targets"
+-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
++    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
++    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
++    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
```
[the remark this answers is not in the document]
Comment (r4ve1) [reply ports/zycore/fix-install.patch]: Done

File: ports/zycore/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS -DZYCORE_BUILD_SHARED_LIB=${ZYCORE_BUILD_SHARED}
+)
+
+vcpkg_cmake_install()
+
+
+vcpkg_cmake_config_fixup(
+    CONFIG_PATH lib/cmake/${PORT}
```
[the remark this answers is not in the document]
Comment (r4ve1) [reply ports/zycore/portfile.cmake]: Done

File: ports/zycore/portfile.cmake [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+
+vcpkg_cmake_config_fixup(
+    CONFIG_PATH lib/cmake/zycore
+)
+    
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+vcpkg_install_copyright(FILE_LIST ${SOURCE_PATH}/LICENSE)
```
Comment (dg0yt) [inline ports/zycore/portfile.cmake:28]: Can you please add quotes to the path expression, and add a line break at the end?

**Review by dg0yt — COMMENTED** [review]
Thanks for the update and the patience.

File: ports/zycore/vcpkg.json (hunk 6 above)
Comment (flobernd) [reply ports/zycore/vcpkg.json:4]: @dg0yt No, the LIBC replacement functions are prefixed and must be used explicitly instead of the original ones.

**Review by dg0yt — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**